### PR TITLE
server: allow to specify DateTime for blocking states

### PR DESCRIPTION
**Pull request by pierre**

### CODE REVIEW

File: profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java [hunk 1]
```diff
@@ -1417,6 +1420,268 @@ public void testSubscriptionEventsWithCatalogChange() throws Exception {
...
+        callbackServlet.pushExpectedEvents(ExtBusEventType.SUBSCRIPTION_PHASE,
+                                           ExtBusEventType.INVOICE_CREATION,
+                                           ExtBusEventType.INVOICE_PAYMENT_SUCCESS,
+                                           ExtBusEventType.PAYMENT_SUCCESS);
+        clock.addDays(30);
+        callbackServlet.assertListenerStatus();
+
+        // Cancel IMM (Billing EOT)
+        callbackServlet.pushExpectedEvents(ExtBusEventType.SUBSCRIPTION_CANCEL,
+                                           ExtBusEventType.ENTITLEMENT_CANCEL);
```
Comment (sbrossie) [inline profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java:1622]: I suppose the `ENTITLEMENT_CANCEL` is for `ActionType.REQUESTED` as the billing cancelation is EOT?

**Review by sbrossie — APPROVED** [review]

File: profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java (hunk 1 above)
Comment (pierre) [reply profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java:1622]: Actually, the entitlement is effective right away (while indeed, billing cancellation is EOT).

Comment (sbrossie) [reply profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java:1622]: I meant: `ExtBusEventType.SUBSCRIPTION_CANCEL` is for `ActionType.REQUESTED` as the billing cancelation is EOT?

Comment (pierre) [reply profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java:1622]: Correct. It's in the JSON metadata of the event:

```
{"actionType":"REQUESTED","bundleExternalKey":"719cddd2-9774-4c87-8621-8aa526c562e7"}
```

**Comment by sbrossie** [comment]
I don't think we have a strong need to have this in `master` because this is tracked as https://github.com/killbill/killbill/issues/1375 for `0-24`, so should we just close?

**Comment by pierre** [comment]
This will be in 0.23.x only.

**Comment by pierre** [comment]
This will be in 0.23.x only.

**Pull request closed without merging**